new/supermarket_namespace_governance.md

```diff
@@ -0,0 +1,167 @@
+
+Some languages, by convention, use a special character for this
+purpose. 'alice/json' vs 'bob/json' is a popular method. Enabling this
+requires coordination among the ecosystem tooling at every level  and
+implementations vary wildly. It would be worth exploring the logistics
+of around this, if for no other reason than to have something to point
+to when people suggest it.
+
+Without a special character, semantic strategies fall on a spectrum.
+At one end there is 'json' vs 'json2', or 'json-alice' vs 'json-bob'.
+At the other end, you have... Ruby. While it may be perfectly
+acceptable in the Ruby community to name an XML processing library
+after a Japanese tree saw, I suspect that the Chef community might
+prefer something less random. A governance activity might include the
+creation and enforcement of a recommended naming conventions.
+   
+Some languages have the notion of "core modules", where different
+governance rules come into play. This feels like something that Chef
+should adopt.
+
+This model starts to show cracks when you consider that most Chef
+cookbooks currently on the supermarket are not written in a reusable
+manner. Cookbooks that provide resources can be considered libraries
+or modules, but recipes are really opinionated state policies.
+
+### Chef as a distribution
+
+In this model, we draw parallels with Linux and Unix distributions.
+Most distributions can be described as the set of available packages
+and the metadata that binds them.
+
+Namespace governance is typically optimized for content curation, with
+an emphasis on interoperability. To achieve this, packaging policies
+are created to enforce common conventions.
+
+This model is often used by sophisticated Chef shops, who often write
+everything themselves and rarely consume cookbooks from the public
+Supermarket.
+   
+In Linux distributions, two packages are rarely, if ever, concerned
+with the same software. Rather than having 'alice/apache' vs
+'bob/apache', the distribution will pick a preferred package and
+declare if "the way of the system". Often these packages vary wildly
+between distributions. The configuration conventions around the Apache
+httpd server is a great example of this.
+
+   
+Distributions almost always have a notion of "core packages", much the
+same as many languages. Alternative or additional packages are
+typically used by enabling extra repository sources. Some  systems
+have quazi-official repositories where packages outside the "core" set
+are maintained, and usually still follow the packaging policy for the
+distribution. The RHEL core/EPEL/SCL systems are examples of this, as
+are the Ubuntu universe / PPA systems.
+
+Governing the Supermarket like a Linux distribution might ultimately
+provide the better experience for consumers, but it requires, by an
+order of magnitude, more effort and thought. Interestingly, the
+current flat namespace and dependency tooling fit this model well.
+      
```

Comment:
It could be really valuable to have something like PPAs so that there doesn't absolutely have to be consensus on how particular problems are solved, but much like a Linux distribution, there can be a default solution that works for most, and easily surfaced alternate implementations.  I think this would also address @coderanger's comment in the main discussion, so that if someone wants to build an opinionated or flavored namespace which depend on each other to provide functionality otherwise typically provided by a 'core' cookbook, authorship and control are as easily established as for a github repository or Launchpad PPA.  The PPA model is also notable because often packages start there and are promoted to the main distribution as additions or replacement for deprecated packages.